src/e_lgamma_r.jl

```diff
@@ -0,0 +1,399 @@
+const t14 =  3.35529192635519073543e-04 #= 0x3F35FD3E, 0xE8C2D3F4 =#
+const u0  = -7.72156649015328655494e-02 #= 0xBFB3C467, 0xE37DB0C8 =#
+const u1  =  6.32827064025093366517e-01 #= 0x3FE4401E, 0x8B005DFF =#
+const u2  =  1.45492250137234768737e+00 #= 0x3FF7475C, 0xD119BD6F =#
+const u3  =  9.77717527963372745603e-01 #= 0x3FEF4976, 0x44EA8450 =#
+const u4  =  2.28963728064692451092e-01 #= 0x3FCD4EAE, 0xF6010924 =#
+const u5  =  1.33810918536787660377e-02 #= 0x3F8B678B, 0xBF2BAB09 =#
+const v1  =  2.45597793713041134822e+00 #= 0x4003A5D7, 0xC2BD619C =#
+const v2  =  2.12848976379893395361e+00 #= 0x40010725, 0xA42B18F5 =#
+const v3  =  7.69285150456672783825e-01 #= 0x3FE89DFB, 0xE45050AF =#
+const v4  =  1.04222645593369134254e-01 #= 0x3FBAAE55, 0xD6537C88 =#
+const v5  =  3.21709242282423911810e-03 #= 0x3F6A5ABB, 0x57D0CF61 =#
+const s0  = -7.72156649015328655494e-02 #= 0xBFB3C467, 0xE37DB0C8 =#
+const s1  =  2.14982415960608852501e-01 #= 0x3FCB848B, 0x36E20878 =#
+const s2  =  3.25778796408930981787e-01 #= 0x3FD4D98F, 0x4F139F59 =#
+const s3  =  1.46350472652464452805e-01 #= 0x3FC2BB9C, 0xBEE5F2F7 =#
+const s4  =  2.66422703033638609560e-02 #= 0x3F9B481C, 0x7E939961 =#
+const s5  =  1.84028451407337715652e-03 #= 0x3F5E26B6, 0x7368F239 =#
+const s6  =  3.19475326584100867617e-05 #= 0x3F00BFEC, 0xDD17E945 =#
+const r1  =  1.39200533467621045958e+00 #= 0x3FF645A7, 0x62C4AB74 =#
+const r2  =  7.21935547567138069525e-01 #= 0x3FE71A18, 0x93D3DCDC =#
+const r3  =  1.71933865632803078993e-01 #= 0x3FC601ED, 0xCCFBDF27 =#
+const r4  =  1.86459191715652901344e-02 #= 0x3F9317EA, 0x742ED475 =#
+const r5  =  7.77942496381893596434e-04 #= 0x3F497DDA, 0xCA41A95B =#
+const r6  =  7.32668430744625636189e-06 #= 0x3EDEBAF7, 0xA5B38140 =#
+const w0  =  4.18938533204672725052e-01 #= 0x3FDACFE3, 0x90C97D69 =#
+const w1  =  8.33333333333329678849e-02 #= 0x3FB55555, 0x5555553B =#
+const w2  = -2.77777777728775536470e-03 #= 0xBF66C16C, 0x16B02E5C =#
+const w3  =  7.93650558643019558500e-04 #= 0x3F4A019F, 0x98CF38B6 =#
+const w4  = -5.95187557450339963135e-04 #= 0xBF4380CB, 0x8C0FE741 =#
+const w5  =  8.36339918996282139126e-04 #= 0x3F4B67BA, 0x4CDAD5D1 =#
+const w6  = -1.63092934096575273989e-03 #= 0xBF5AB89D, 0x0B9E43E4 =#
+
+# Matches OpenLibm behavior exactly, including return of sign
+function _lgamma_r(x::Float64)
+    u = reinterpret(UInt64, x)
+    hx = (u >>> 32) % Int32
+    lx = u % Int32
+
+    #= purge off +-inf, NaN, +-0, tiny and negative arguments =#
+    signgamp = Int32(1)
+    ix = signed(hx & 0x7fffffff)
+    ix ≥ 0x7ff00000 && return x * x, signgamp
+    ix | lx == 0 && return 1.0 / 0.0, signgamp
+    if ix < 0x3b900000 #= |x|<2**-70, return -log(|x|) =#
+        if hx < 0
+            signgamp = Int32(-1)
+            return -log(-x), signgamp
+        else
+            return -log(x), signgamp
+        end
+    end
+    if hx < 0
+        ix ≥ 0x43300000 && return 1.0 / 0.0, signgamp #= |x|>=2**52, must be -integer =#
```

Comment:
I would remove this branch. it isn't free and only helps for numbers less than -2^52.

Reply:
I think this should be reasonably safe. The subsequent `sinpi` and comparison to `0.0` will ensure an early exit, but it does cost ~1ns. It seems reasonable to assume that values less than -2^52 would be uncommon.